File: js/emit_power.js
function
emit_power(u, p) // p = y^x
{
	if (cadr(p) == symbol(EXP1)) {
		emit_roman_text(u, "exp");
		emit_subexpr(u, caddr(p));
		return;
	}

	if (isimaginaryunit(p)) {
		if (isimaginaryunit(get_binding(symbol(SYMBOL_J)))) {
			emit_italic_text(u, "j");
			return;
		}
		if (isimaginaryunit(get_binding(symbol(SYMBOL_I)))) {
			emit_italic_text(u, "i");
			return;
		}
	}

	if (isnegativenumber(caddr(p))) {
		emit_reciprocal(u, p);
		return;
	}

	emit_base(u, cadr(p));
	emit_exponent(u, caddr(p));
}

File: app/emit_power.c
#include "app.h"

void
emit_power(struct atom *p)
{
	if (cadr(p) == symbol(EXP1)) {
		emit_roman_string("exp");
		emit_args(cdr(p));
		return;
	}

	if (isimaginaryunit(p)) {
		if (isimaginaryunit(get_binding(symbol(SYMBOL_J)))) {
			emit_italic_char('j');
			return;
		}
		if (isimaginaryunit(get_binding(symbol(SYMBOL_I)))) {
			emit_italic_char('i');
			return;
		}
	}

	if (isnegativenumber(caddr(p))) {
		emit_reciprocal(p);
		return;
	}

	emit_base(cadr(p));
	emit_exponent(caddr(p));
}
